src/components/data_portal/DataPortalShared.tsx

```diff
@@ -210,32 +231,158 @@ export const standaloneStyle = makeStyles((theme: Theme) =>
 )
 
 //given a valid researcher, study, or participant id,
-//an array of ids is returned
-export async function generate_ids(id_set) {
+//an array of participant ids is returned
+export async function generate_ids(id_set, return_with_names = false) {
   if (typeof id_set === "string") {
     let { data: parents } = await LAMP.Type.parent(id_set)
     //if study exists, return id_set
-    if (parents?.Study) return [id_set]
+    if (parents?.Study) {
+      if (return_with_names) {
+        let result = await LAMP.Type.getAttachment(id_set, "lamp.name")
+        return { [id_set]: result["data"] ? result["data"] : null }
+      } else return [id_set]
+    }
     //else if researcher exists
     //this is a study
     else if (parents?.Researcher) {
       let res = await LAMP.Participant.allByStudy(id_set)
-      return res.map((participant) => participant.id)
+      if (return_with_names) {
+        let resObj = (await Promise.all(
+          res.map((part) => {
+            return LAMP.Type.getAttachment(part.id, "lamp.name").then((result) => {
```

Comment:
Can you refactor **ALL** API calls made by the data portal to use the `sonata`/query transform endpoint instead?